### DISCUSSION TITLE
Problem with binarySource="install" and tools from local node_modules

### DISCUSSION THREAD

**Discussion by aizerin** [renovatebot/renovate#25251] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.11.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I recently upgraded from renovate 32.159.5 to 37.11.1 and I started to have issues with JavaScript projects when I was relying on tools from projects node_modules (like husky, prettier, eslint, ...). I was digging more into this issue and I found that the problem is with the change of binarySource from "global" to "install". If I use global with full docker image then it works as before, but I like the slim version because it is faster than before.

It seems that renovate will not use the node_modules/.bin from installed project directory to resolve installed tools.

I tried to create a minimal example but I was not successful. When I use your renovate hosting then it works without any issues. https://github.com/aizerin/renovate_minimal . But I noticed from the logs that you are using binarySource="docker".

So here is the example of problematic parts in my configuration:

package.json
```
{
	"scripts": {
		"prepare": "husky install", -->> this will generate /bin/sh: 1: husky: not found\nerror Command failed with exit code 127.\n
	},
}
```

renovate.json
```
{
  "postUpgradeTasks": {
    "commands": [
      "awk script to create changelog", -->> awk command works correctly
      "yarn run prettier CHANGELOG.md --write" -->> this will fail same as husky
    ],
    "fileFilters": ["CHANGELOG.md"]
  },
 "ignoreScripts": false,
}
```

I'm running renovate on a private k8s cluster with this helm config.

```
renovate:
  cronjob:
    schedule: "*/30 6-22/2 * * *"
    concurrencyPolicy: "Replace"
    successfulJobsHistoryLimit: 1
    failedJobsHistoryLimit: 1

  env:
    LOG_LEVEL: "debug"

  resources:
    requests:
      cpu: "3"
      memory: "5Gi"
    limits:
      cpu: "3"
      memory: "5Gi"

  renovate:
    config: |
      {
        "repositories": [a lot of repositoryes],
        "platform": "bitbucket-server",
        "commitMessagePrefix": "RENOVATE: ",
        "gitAuthor": "Renovate",
        "allowPostUpgradeCommandTemplating": true,
        "allowedPostUpgradeCommands": [
            "^awk.*$",
            "^mv.*$",
            "^npx @gef-ui.*$",
            "^yarn run prettier.*$"        ],
        "allowCustomCrateRegistries": true,
        "allowScripts": true,
        "allowPlugins": true,
        "exposeAllEnv": true
      }
    persistence:
      cache:
        enabled: true
        storageClass: "longhorn"
        storageSize: "8Gi"
```

### Relevant debug logs

<details><summary>Logs</summary>

```
"message": "Command failed: yarn install --ignore-engines --ignore-platform --network-timeout 100000\n/bin/sh: 1: husky: not found\nerror Command failed with exit code 127.\n",

"stderr": "/bin/sh: 1: husky: not found\nerror Command failed with exit code 127.\n"

"stderr": "/bin/sh: 1: husky: not found\nerror Command failed with exit code 127.\nerror Command \"eslint\" not found.\nerror Command \"prettier\" not found.\n"
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by viceice** [comment]
are you seeing this message inside logs?

https://github.com/renovatebot/renovate/blob/939c1c70c988c1d50b5cccac35cb2257d127e4e2/lib/modules/manager/npm/post-update/yarn.ts#L155

**Reply by aizerin** [reply]
Yes I can see it, so the problem is that node_modules are not installed. I can change prettier in postUpgradeTasks to run with npx instead of local installation. But the biggest issue is with the husky in prepare script `"prepare": "husky install"`. I'm not able to change 50+ projects in our codebase. Is it possible to preinstall this binary globally without creating my own docker image or init container?

**Reply by viceice** [reply]
set `skipInstalls=false` and you should get your node_modules
